gobblin-kafka/src/main/java/gobblin/kafka/writer/KafkaDataWriter.java

```diff
@@ -0,0 +1,127 @@
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.kafka.writer;
+
+import java.io.IOException;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.writer.DataWriter;
+
+
+/**
+ * Implementation of KafkaWriter that wraps a {@link KafkaProducer}.
+ * This does not provide transactional / exactly-once semantics.
+ * Applications should expect data to be possibly written to Kafka even if the overall Gobblin job fails.
+ *
+ */
+@Slf4j
+public class KafkaDataWriter<D> implements DataWriter<D> {
+
+  private final KafkaProducer<String, D> producer;
+  private final String topic;
+  private final AtomicInteger recordsWritten;
+  private final AtomicInteger bytesWritten;
+
+
+  private static final String KEY_SERIALIZER_CONFIG = "key.serializer";
+  private static final String DEFAULT_KEY_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+  private static final String VALUE_SERIALIZER_CONFIG = "value.serializer";
+  private static final String DEFAULT_VALUE_SERIALIZER = "org.apache.kafka.common.serialization.ByteArraySerializer";
+
+  public KafkaDataWriter(Properties props) {
+    recordsWritten = new AtomicInteger(0);
+    bytesWritten = new AtomicInteger(0);
+
+    Config config = ConfigFactory.parseProperties(props);
+    this.topic = config.getString(KafkaWriterConfigurationKeys.KAFKA_TOPIC);
+    Properties producerProperties = stripPrefix(props, KafkaWriterConfigurationKeys.KAFKA_PRODUCER_CONFIG_PREFIX);
+
+    //Provide default properties if not set from above
+    if (!producerProperties.containsKey(KEY_SERIALIZER_CONFIG)) {
+      producerProperties.setProperty(KEY_SERIALIZER_CONFIG, DEFAULT_KEY_SERIALIZER);
+    }
+    if (!producerProperties.containsKey(VALUE_SERIALIZER_CONFIG)) {
+      producerProperties.setProperty(VALUE_SERIALIZER_CONFIG, DEFAULT_VALUE_SERIALIZER);
+    }
+    //producerProperties.put("request.required.acks", "1");
+    this.producer = new KafkaProducer<>(producerProperties);
+  }
+
+  private Properties stripPrefix(Properties props, String prefix) {
+    Properties strippedProps = new Properties();
+    int prefixLength = prefix.length();
+    for (String key: props.stringPropertyNames())
+    {
+      if (key.startsWith(prefix))
+      {
+        strippedProps.setProperty(key.substring(prefixLength), props.getProperty(key));
+      }
+    }
+    return strippedProps;
+  }
+
+
+  @Override
+  public void write(D record)
+      throws IOException {
+    this.producer.send(new ProducerRecord<String, D>(topic, record), new Callback() {
```

Comment:
Is it necessary to create a new `Callback` object for each record written. Since the class doesn't keep any state, can you just cache the object.